app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/DrawerViewModel.java

```diff
@@ -0,0 +1,128 @@
+import com.teamagam.gimelgimel.domain.map.DisplayVectorLayersInteractor;
+import com.teamagam.gimelgimel.domain.map.DisplayVectorLayersInteractorFactory;
+import com.teamagam.gimelgimel.domain.map.SetVectorLayerVisibilityInteractorFactory;
+
+import javax.inject.Inject;
+
+@PerActivity
+public class DrawerViewModel extends BaseViewModel
+        implements DisplayVectorLayersInteractor.Displayer {
+
+    @Inject
+    DisplayVectorLayersInteractorFactory mDisplayVectorLayersInteractorFactory;
+    @Inject
+    SetVectorLayerVisibilityInteractorFactory mSetVectorLayerVisibilityInteractorFactory;
+
+    private DrawerStateListener mDrawerStateListener;
+
+    private DisplayVectorLayersInteractor mDisplayVectorLayersInteractor;
+
+    private DrawerLayout mDrawerLayout;
+
+    private StringToIntegerMapper mStringToIntegerMapper;
+
+    @Inject
+    public DrawerViewModel() {
+    }
+
+    public void start() {
+        mStringToIntegerMapper = new StringToIntegerMapper();
+        mDrawerStateListener = new DrawerStateListener();
+
+        mDisplayVectorLayersInteractor = mDisplayVectorLayersInteractorFactory.create(this);
+    }
+
+    public void setDrawerMenuSelectedListenerTo(NavigationView navigationView) {
+        navigationView.setNavigationItemSelectedListener(
+                new DrawerItemSelectedListener());
+    }
+
+    public void setDrawerLayout(DrawerLayout drawerLayout) {
+        mDrawerLayout = drawerLayout;
+    }
+
+    @Override
+    public void displayShown(VectorLayerPresentation vectorLayerPresentation) {
+        display(vectorLayerPresentation, true);
+    }
+
+    @Override
+    public void displayHidden(VectorLayerPresentation vectorLayerPresentation) {
+        display(vectorLayerPresentation, false);
+    }
+
+    private void display(VectorLayerPresentation vectorLayerPresentation, boolean isVisible) {
+        if (!mStringToIntegerMapper.contains(vectorLayerPresentation.getId())) {
+            int intId = mStringToIntegerMapper.put(vectorLayerPresentation.getId());
+            ((MainActivityDrawer) mView).addToMenu(vectorLayerPresentation.getName(), intId);
+        }
+        ((MainActivityDrawer) mView).setChecked(
+                mStringToIntegerMapper.get(vectorLayerPresentation.getId()), isVisible);
+    }
+
+    public void resume() {
+        mDisplayVectorLayersInteractor.execute();
+        mDrawerLayout.addDrawerListener(mDrawerStateListener);
+    }
+
+    public void pause() {
+        mDisplayVectorLayersInteractor.unsubscribe();
+        mDrawerLayout.removeDrawerListener(mDrawerStateListener);
+    }
+
+    private class DrawerStateListener implements DrawerLayout.DrawerListener {
+        @Override
+        public void onDrawerSlide(View drawerView, float slideOffset) {
+
+        }
+
+        @Override
+        public void onDrawerOpened(View drawerView) {
+            sLogger.userInteraction("Drawer opened");
+
+            ((MainActivityDrawer) mView).setNavHeaderText(drawerView);
```

Comment:
The reason is that username wasn't immutable until now. This branch doesn't yet know about the new status of the username.